backend/dataall/tasks/data_sharing/share_managers/lf_share_manager.py

```diff
@@ -0,0 +1,551 @@
+            database=self.shared_db_name,
+            tablename=table.GlueTableName,
+        ):
+            logger.info(
+                f'Resource link could not be found '
+                f'on {self.target_environment.AwsAccountId}/{self.shared_db_name}/{table.GlueTableName} '
+                f'skipping revoke actions...'
+            )
+            return True
+
+        logger.info(
+            f'Revoking resource link access '
+            f'on {self.target_environment.AwsAccountId}/{self.shared_db_name}/{table.GlueTableName} '
+            f'for principal {self.env_group.environmentIAMRoleArn}'
+        )
+        LakeFormation.batch_revoke_permissions(
+            SessionHelper.remote_session(self.target_environment.AwsAccountId).client(
+                'lakeformation', region_name=self.target_environment.region
+            ),
+            self.target_environment.AwsAccountId,
+            [
+                {
+                    'Id': str(uuid.uuid4()),
+                    'Principal': {
+                        'DataLakePrincipalIdentifier': self.env_group.environmentIAMRoleArn
+                    },
+                    'Resource': {
+                        'Table': {
+                            'DatabaseName': self.shared_db_name,
+                            'Name': table.GlueTableName,
+                            'CatalogId': self.target_environment.AwsAccountId,
+                        }
+                    },
+                    'Permissions': ['DESCRIBE'],
+                }
+            ],
+        )
+        return True
+
+    def revoke_source_table_access(self, table):
+        """
+        Revokes access to the source glue table
+        Parameters
+        ----------
+        table : models.DatasetTable
+
+        Returns
+        -------
+        True if revoke is successful
+        """
+        if not Glue.table_exists(
+            accountid=self.target_environment.AwsAccountId,
+            region=self.target_environment.region,
+            database=self.shared_db_name,
+            tablename=table.GlueTableName,
+        ):
+            logger.info(
+                f'Source table could not be found '
+                f'on {self.source_environment.AwsAccountId}/{self.dataset.GlueDatabaseName}/{table.GlueTableName} '
+                f'skipping revoke actions...'
+            )
+            return True
+
+        logger.info(
+            f'Revoking source table access '
+            f'on {self.source_environment.AwsAccountId}/{self.dataset.GlueDatabaseName}/{table.GlueTableName} '
+            f'for principal {self.env_group.environmentIAMRoleArn}'
+        )
+        LakeFormation.revoke_source_table_access(
+            target_accountid=self.target_environment.AwsAccountId,
+            region=self.target_environment.region,
+            source_database=self.dataset.GlueDatabaseName,
+            source_table=table.GlueTableName,
+            target_principal=self.env_group.environmentIAMRoleArn,
```

Comment:
Should we be using the share principal IAM Role here as the `target_principal` rather than EnvGroup IAM Role

Reply:
Yes, during the testing I also ran into this issue when revoking to principals that are not an environment admin group (e.g. when testing consumption roles). O started working on a fix yesterday